kube-runtime/src/reflector/object_ref.rs

```diff
@@ -28,10 +31,11 @@ use std::{
 ///     ObjectRef::<Secret>::new("a").erase(),
 /// );
 /// ```
+#[non_exhaustive]
 pub struct ObjectRef<K: Resource> {
-    dyntype: K::DynamicType,
+    pub dyntype: K::DynamicType,
     /// The name of the object
-    pub name: String,
+    pub name: Option<String>,
```

Comment:
This is a minor breakage, but it allows us to get rid of the panic in `from_obj`.

Reply:
Actually, screw it, I'll revert this change for now and leave it as a future improvement to figure out...

Reply:
ah, ok. yeah, i imagine this might have some implications down the line (like maybe we have to ignore those types of watch events somewhere)